### DISCUSSION TITLE
glucose readings received by followers from the master in the form of three question marks

### DISCUSSION THREAD

**Discussion by Inkalika** [NightscoutFoundation/xDrip#3050] [Bug reports]
When calibrations are reset in the wizard via the tab: Stop sensor - removing calibrations only, glucose readings are no longer transmitted on followers, the current glucose value is displayed as three red question marks, but readings for carbohydrates eaten and insulin injected continue to be accepted by followers. Reinstalling applications from followers does not help.
![Screenshot_20230827-135227_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/89869109/9e6ef170-b34a-468c-abc6-4370a0dc3f54)
![Screenshot_20230827-135246_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/89869109/2c7714d4-6e42-4706-9023-e0affd681f28)
![Screenshot_20230827-135250_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/89869109/02b7e06a-4464-4040-b5f8-db85edea6433)
![Screenshot_20230827-135555_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/89869109/9f2ba9e4-3164-4514-9c60-625717e55540)

**Comment by Navid200** [comment]
It's not clear what you are describing.  What does the master show when the follower shows the question marks?  That's very important.  But, you are not covering that.

You can put the master and follower phones side by side and take a single screenshot showing the two phones at the same time so that we can see the time on each phone.

**Comment by Inkalika** [comment]
![1](https://github.com/NightscoutFoundation/xDrip/assets/89869109/1ea5d013-84c7-4a3d-af63-cf3e7ccb250c)
![2](https://github.com/NightscoutFoundation/xDrip/assets/89869109/fffb535d-2184-45fd-a678-a8360415f97a)
![3](https://github.com/NightscoutFoundation/xDrip/assets/89869109/c06c521d-3c3c-47b5-8010-67a0bfc877e7)

**Reply by Navid200** [reply]
When you view the hardware data source page on the follower, does it show xDrip sync follower?
https://navid200.github.io/xDrip/docs/HardwareDataSource.html

**Reply by Inkalika** [reply]
This situation occurs on both followers (Samsung Galaxy M51 (Android 10), Redmy 12C (Android 11) and replacing the application on followers with other versions does not help. I assume that this is related to the master (but the setting on the master's phone is to be a master for followers enabled). Most likely, reinstalling the application on the Master will help, but for now I hope to solve this problem in other ways - maybe you need to enable some kind of check mark on the Master or Follower
![Screenshot_20230829-072112_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/89869109/a3c1927f-ba13-4223-99b5-9e136c195c5d)

**Reply by Navid200** [reply]
Will it resolve if you restart the follower phone?

**Reply by Inkalika** [reply]
yes, I restarted the phone, restarted the collector, deleted data from the future, forgot the sensor and re-uploaded the Master settings to the follower via a QR code. But it didn't help.

**Reply by Navid200** [reply]
Why would you load the follower with master settings?
If I understand what you are describing, this is a follower problem.  
Basically, for us to be able to help with a bug report, we need to reproduce the problem.
I have tried to understands the problem.  But, I am sorry to say you have not made it very easy.

**Reply by Inkalika** [reply]
There are two followers. The last time I noticed that immediately after resetting the calibrations on the Master, on the follower, where calibrations and glucose display were not enabled, taking into account calibrations (the standard xdrip calibration plugin is enabled, the calibrated sugar display switches are enabled), question marks appeared instead of glucose. But on the second follower, the testimony continued to go. It had exactly the same calibration settings as on the Master. And after I turned off the calibration plugin on this follower, the display of calibrated sugar, question marks also appeared there. And after restoring the calibration settings on the follower, the readings have not changed. I understand that this has something to do with the calibration settings of the Master and Followers. I can attach the Followers and Master settings later.

**Reply by Navid200** [reply]
Great, that is very useful.
I will have to ask someone else to have a look.

This is fundamentally problematic if we allow xDrip to calibrate the follower.
I wouldn't understand any logic that would justify that.
I understand if we allow a follower to calibrate the master.  But, the follower should only show what the master shows.  There should be no calibration on the follower.

But, this is only my expectation.  I cannot tell you that's what xDrip does.  So, I will ask someone else to help here.

**Reply by Inkalika** [reply]
Initially, the master was one of the Followers (just the one on which the calibration settings remained). Later, a separate Master phone was purchased, and the old Master became a Follower, because the calibration settings were preserved. I did not try to calibrate the Master with a Follower and this option was disabled in the Master. I see the problem precisely in the fact that if calibration options were enabled in the Follower at the time of connection, which coincide with the settings in the Master, then after deleting them in the Follower or clearing all calibrations (possibly stopping the sensor and then turning it on again), the glucose display in the Followers fails. So far, the solution is to disable all calibration options in the Followers, and in the Master no longer delete all calibrations through the Stop Sensor window (there are no problems when deleting calibrations through the Table)

**Reply by Inkalika** [reply]
Follower settings
![Screenshot_20230829-084948_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/89869109/7a78c6de-7d94-4421-8708-96145696dbe6)

**Reply by Inkalika** [reply]
Master settings
![IMG-20230829-WA0002](https://github.com/NightscoutFoundation/xDrip/assets/89869109/03c904e4-d105-47ae-8d22-120fe11ce445)

**Reply by Navid200** [reply]
I don't know why you have done it this way.  I can tell you what I would do in your place.  But, it's up to you what you want to do.

There can be only one master.
Regardless of which phone you use as the master, it should be the one that has calibrations on it.

The follower should be set up as shown here:  https://youtu.be/LcgjfbYcWkE?si=jiGziGqkDaZkH1Al
Just that and nothing else.  There should be no calibrations on the follower.

**Reply by Inkalika** [reply]
Thank you. Calibration via Followers was not carried out, but the plugin for displaying calibrated glucose was enabled on one of the Followers. But on the other, these options were not enabled and the transmission of glucose readings broke down precisely when cleaning all calibrations in the Master via Stop sensor.

Conclusion - do not include the calibration plugin in the Followers, display of calibrated readings, because after they are turned off, the readings break. The only thing I don't understand. why do the readings break on the Follower when you clean all the calibrations in the Master.

**Reply by Navid200** [reply]
When you changed the old master to become a follower, did you disable "be master for followers" on it on the xDrip Sync settings page?

**Reply by Inkalika** [reply]
yes, the "be master for followers" option was immediately turned off

**Reply by Navid200** [reply]
OK

Stop all calibration plugins on the followers.
Never ever calibrate a follower.

xDrip sync is used by thousands of people all over the world.   This is another page describing how to enable it:
https://navid200.github.io/xDrip/docs/xDripSync.html

**Reply by Inkalika** [reply]
the screenshot shows that the raw values come, but are not displayed in mmol
![Screenshot_20230829-093138_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/89869109/2cde6e55-dfaf-4ea2-bb61-1b0d165eb0c1)

**Comment by Navid200** [comment]
@tzachi-dar 
Why would xDrip allow the follower to calibrate the follower?
I can think of really dangerous situation that could lead to.
Let's say the child is at school.  The follower parent calibrates the follower to show a higher reading by mistake.  If xDrip allows that, now the parent sees a reasonable reading while the child is actually experiencing hypoglycemia.  

I would call that a disaster.  I would fix that and remove the possibility by never allowing a follower to calibrate the follower.

**Comment by Inkalika** [comment]
Hello. On the current issues, I made the following conclusions: 1) I have a sibioniks sensor, and xdrip collects information from it through a Data Source - 640G 2) The sensor is calibrated only when the plug-in is enabled (Datricsae, xDrip Original, and so on). If the plugin is set to None mode, then calibrations are not accepted. 3) Due to the use of the calibration plugin for this sensor, when two followers are turned on, any changes in calibrations lead to errors in the transmission of glucose levels - it is transmitted in the form of three questions. But if you connect only one follower, then there is no such error. 4) If you connect a night Scout, then sugar data is transmitted without taking into account the calibration plugin, raw data. 5) It is possible to create a driver in data sourse specifically for this sensor, so as not to include an additional calibration plugin, which will solve problems with data transmission to Night Scout and drip followers.

**Comment by Inkalika** [comment]
I have now decided to replace the sensor. I have only one follower left, because when I connect the second one, the readings stop coming on both follower (I get question marks instead of sugar). I did a reset of all calibrations on the master, then stop sensor and start sensor. On the master the readings went. The follower only got 3 new readings and started getting question marks instead of sugar values again. I think it has to do with resetting all calibrations on the master, because as soon as I do it on the sensor that is running, I get the same thing. So the problem is related to errors in the active calibration plugins. When the active calibration plugin is enabled, the disabled "Rare Calibration" checkbox also doesn't work - old calibrations become inactive, the "Rewrite History" checkbox doesn't work - all values for all time change their values after calibration, the transfer of calibrated values to NiteScout doesn't work, they come raw before edits of the active calibration plugin.

**Reply by Inkalika** [reply]
![photo_2023-09-22_06-56-15](https://github.com/NightscoutFoundation/xDrip/assets/89869109/1b61a0c4-eead-42c6-a05f-bb30c4278a66)
It's follower

**Comment by Inkalika** [comment]
![photo_2023-09-22_07-36-54](https://github.com/NightscoutFoundation/xDrip/assets/89869109/db37112d-c58f-4122-ac4c-21b3cb9b3810)
No calibrated sugar levels, therapy comes in. In the glucose value table, raw glucose data comes in

**Comment by Inkalika** [comment]
After performing the initial calibration. the readings appeared on the follower
![photo_2023-09-22_09-36-45](https://github.com/NightscoutFoundation/xDrip/assets/89869109/4b099066-1f39-4c47-b757-5e8e8f72a178)
